packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api
+
+import com.amazonaws.amplify.amplify_api.types.FlutterApiErrorMessage
+import com.amplifyframework.api.ApiCategory
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.GsonVariablesSerializer
+import com.amplifyframework.api.graphql.GraphQLOperation
+import com.amplifyframework.api.graphql.GraphQLRequest
+import com.amplifyframework.api.graphql.GraphQLResponse
+import com.amplifyframework.api.graphql.SimpleGraphQLRequest
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+
+@RunWith(RobolectricTestRunner::class)
+class GraphQLApiUnitTests {
+    lateinit var flutterPlugin: AmplifyApiPlugin
+    private var mockApi = mock(ApiCategory::class.java)
+    private val mockResult: MethodChannel.Result = mock(MethodChannel.Result::class.java)
+    private val mockGraphQLOperation = mock(GraphQLOperation::class.java)
+
+    @Before
+    fun setup() {
+        flutterPlugin = AmplifyApiPlugin()
+        setFinalStatic(Amplify::class.java.getDeclaredField("API"), mockApi)
```

Comment:
#270 and #272 show how to avoid this. Mock the behavior of the native plugin, instead. Leave the category alone.

You may need to open up a second constructor in your `AmplifyApiPlugin`. It should call onto `ApiCategoryBehavior`, instead of `Amplfy.API` directly.